api/src/main/java/org/openmrs/validator/DiagnosisValidator.java

```diff
@@ -0,0 +1,67 @@
+package org.openmrs.validator;
+
+import org.openmrs.Diagnosis;
+import org.openmrs.Encounter;
+import org.openmrs.api.APIException;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+import org.springframework.validation.Validator;
+
+public class DiagnosisValidator implements Validator {
+
+	/**
+	 * @should support Diagnosis class
+	 */
+	@Override
+	public boolean supports(Class<?> aClass) {
+		return Diagnosis.class.isAssignableFrom(aClass);
+	}
+
+	/**
+	 * @see org.springframework.validation.Validator#validate(java.lang.Object,
+	 *      org.springframework.validation.Errors)
+	 * @should fail validation if rank is null
+	 * @should fail validation if certainty is null
+	 * @should fail validation if diagnosis is null
+	 * @should fail validation if diagnosisId is null
+	 * @should fail for a null object
+	 */
+	@Override
+	public void validate(Object o, Errors errors) {
+		Diagnosis diagnosis = (Diagnosis)o;
+		
+		if(diagnosis == null){
```

Comment:
Did you run mvn clean install before committing? If you did, this file would have been formatted differently